aws/resource_aws_rds_cluster_test.go

```diff
@@ -3518,3 +3546,100 @@ resource "aws_rds_cluster" "test" {
 }
 `, rName, enableHttpEndpoint)
 }
+
+func testAccAWSRDSClusterConfig_MultipleClustersInGlobalCluster(rNameGlobal, rNamePrimary, rNameSecondary string) string {
+	return composeConfig(
+		testAccMultipleRegionProviderConfig(2),
+		fmt.Sprintf(`
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "alternate" {
+  provider = "awsalternate"
+  state = "available"
+
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
+resource "aws_rds_global_cluster" "test" {
+  global_cluster_identifier = "%[1]s"
+  engine                    = "aurora-mysql"
+  engine_version            = "5.7.mysql_aurora.2.07.1"
+}
+
+resource "aws_rds_cluster" "primary" {
+  cluster_identifier              = "%[2]s"
+  database_name                   = "mydb"
+  master_username                 = "foo"
+  master_password                 = "barbarbar"
+  skip_final_snapshot             = true
+  global_cluster_identifier       = aws_rds_global_cluster.test.id
+  engine                          = "aurora-mysql"
+  engine_version                  = "5.7.mysql_aurora.2.07.1"
+}
+
+resource "aws_rds_cluster_instance" "primary" {
+  identifier         = "%[2]s"
+  cluster_identifier = aws_rds_cluster.primary.id
+  instance_class     = "db.r4.large" # only db.r4 or db.r5 are valid for Aurora global db
+  engine             = "aurora-mysql"
+  engine_version     = "5.7.mysql_aurora.2.07.1"
+}
+
+resource "aws_vpc" "alternate" {
+  provider   = "awsalternate"
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = "%[3]s"
+  }
+}
+
+resource "aws_subnet" "alternate" {
+  provider          = "awsalternate"
+  count             = 3
+  vpc_id            = aws_vpc.alternate.id
+  availability_zone = data.aws_availability_zones.alternate.names[count.index]
+  cidr_block        = "10.0.${count.index}.0/24"
+
+  tags = {
+    Name = "%[3]s"
+  }
+}
+
+resource "aws_db_subnet_group" "alternate" {
+  provider   = "awsalternate"
+  name       = "%[3]s"
+  subnet_ids = aws_subnet.alternate[*].id
+}
+
+resource "aws_rds_cluster" "secondary" {
+  provider                      = "awsalternate"
+  cluster_identifier            = "%[3]s"
+  db_subnet_group_name          = aws_db_subnet_group.alternate.name
+  skip_final_snapshot           = true
+  source_region                 = data.aws_region.current.name
+  global_cluster_identifier     = aws_rds_global_cluster.test.id
+  engine                        = "aurora-mysql"
+  engine_version                = "5.7.mysql_aurora.2.07.1"
+  depends_on                    = [aws_rds_cluster_instance.primary]
+
+  lifecycle {
+    ignore_changes = [
+      replication_source_identifier,
+    ]
+  }
```

Comment:
I'm curious about this, is the API automatically adding it? 😄  If so, we may want to mark it as `Computed: true` in a separate change.

Reply:
in the original issue's description, i didn't see a workaround to this behavior so i thought maybe the test configuration is a little off? took me a bit of experimenting to get the right combination of primary/secondary rds_clusters 😅

Reply:
Interesting! For what its worth, I think we're going to mark `global_cluster_identifier` as `Computed: true` anyways for https://github.com/terraform-providers/terraform-provider-aws/issues/10965

Reply:
nice, that issue def looks similar to what i ran into here

Reply:
ah looks like there's already an issue related to this block: https://github.com/terraform-providers/terraform-provider-aws/issues/10150